src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageEndpoint.cs

```diff
@@ -80,15 +79,15 @@ internal class RazorLanguageEndpoint : IRazorLanguageQueryHandler, IRazorMapToDo
 
         public async Task<RazorLanguageQueryResponse> Handle(RazorLanguageQueryParams request, CancellationToken cancellationToken)
         {
-            long documentVersion = -1;
+            int? documentVersion = null;
             DocumentSnapshot documentSnapshot = null;
             await Task.Factory.StartNew(() =>
             {
                 _documentResolver.TryResolveDocument(request.Uri.GetAbsoluteOrUNCPath(), out documentSnapshot);
                 if (!_documentVersionCache.TryGetDocumentVersion(documentSnapshot, out documentVersion))
                 {
                     // This typically happens for closed documents.
-                    documentVersion = UndefinedDocumentVersion;
```

Comment:
I'd verify that this doesn't break VSCode. I can't recall exactly why we added a specific UndefinedDocumentVersion but VSCode rings a bell in my memory

Reply:
I still get Hovers, Semantic tokens, signatureHelpers, Completiongs, and nothing angry-looking in the log.